### ISSUE TITLE
[rancherd] Bump RKE2 version - 1.20

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#31375]
Once the Rancher team is ready, we should plan to update rancherd to RKE2 version 1.20.x (latest stable).

**Comment by davidnuzik** [comment]
Related issue for 1.19.x: https://github.com/rancher/rancher/issues/30492 - this issue has more context and and example covering the processs if you look at the linked PR and comments.

**Comment by davidnuzik** [comment]
Utilize v1.20.5+k3s1 now instead of v1.20.4+k3s1 due to a CVE called out in v1.20.5 k3s release notes.

**Comment by bmdepesa** [comment]
Rancherd is failing to start. It looks to be using the `rke2-runtime:latest` tag instead of `rke2-runtime:v1.20.6-rke2r1`

```
> curl -sfL https://get.rancher.io | sudo INSTALL_RANCHERD_VERSION=v2.5.8-rc12 sh -
> systemctl enable rancherd-server.service && systemctl start rancherd-server.service
> journalctl -eu rancherd-server -f
...
rancherd[2608]: time="2021-04-28T16:43:49Z" level=fatal msg="Runtime image index.docker.io/rancher/rke2-runtime:latest is not a not a reference to a digest or version tag matching pattern ^v[0-9]"
...
```

**Comment by bmdepesa** [comment]
rancher/rancher:v2.5.8-rc15

Testing manually with the possible workaround we see another issue where the `rancher` and `rancher-namespace` manifests are not included in the RancherD installation, and Rancher never deploys. 

In v2.5.8-rc15 RancherD installation:
```
/var/lib/rancher/rke2/server/manifests$ ls
rke2-canal.yaml  rke2-coredns.yaml  rke2-ingress-nginx.yaml  rke2-kube-proxy.yaml  rke2-metrics-server.yaml
```

vs. v2.5.7 installation:
```
/var/lib/rancher/rke2/server/manifests# ls
rancher-namespace.yaml  rke2-canal.yaml    rke2-ingress-nginx.yaml  rke2-metrics-server.yaml
rancher.yaml            rke2-coredns.yaml  rke2-kube-proxy.yaml
```

**Comment by rancher-max** [comment]
Validated rancherd is using v1.20.5+rke2r1 and installs correctly in both airgap and not.
